### PR TITLE
Bug fixes for directed edge

### PR DESCRIPTION
**Pull request by isaacbrodsky**
Based upon #553. This updates the tests with some cases found via fuzzing to the test suite.

### CODE REVIEW

File: src/apps/testapps/testGridDisk.c [hunk 1]
```diff
@@ -329,6 +329,19 @@ SUITE(gridDisk) {
         t_assert(out == origin, "Moving to self goes to self");
     }
 
+    TEST(h3NeighborRotations_invalid) {
+        // This is undefined behavior, but it's helpful for it to make sense.
+        H3Index origin = 0x811d7ffffffffffL;
```
Comment (dfellis) [inline src/apps/testapps/testGridDisk.c:334]: Anything special about this particular H3 index?

**Review by dfellis — APPROVED** [review]

File: src/apps/testapps/testGridDisk.c (hunk 1 above)
Comment (isaacbrodsky) [reply src/apps/testapps/testGridDisk.c:334]: No - I just needed a known good index as I wanted the only rotations to be under test.

Comment (nrabinowitz) [reply src/apps/testapps/testGridDisk.c:334]: For my info, is there any difference between indexes with the `L` suffix or without? Should we be consistent?

File: src/h3lib/lib/algos.c [hunk 2]
```diff
@@ -335,6 +335,9 @@ H3Error h3NeighborRotations(H3Index origin, Direction dir, int *rotations,
                             H3Index *out) {
     H3Index current = origin;
 
+    if (dir < CENTER_DIGIT || dir >= INVALID_DIGIT) {
```
Comment (nrabinowitz) [inline src/h3lib/lib/algos.c:338]: For my info, what does `CENTER_DIGIT` even do? Returns the current index?

**Review by nrabinowitz — APPROVED** [review]

File: src/apps/testapps/testGridDisk.c (hunk 1 above)
Comment (isaacbrodsky) [reply src/apps/testapps/testGridDisk.c:334]: All valid H3 indexes would only be valid integers as 64 bit, so no I don't believe so.

File: src/h3lib/lib/algos.c (hunk 2 above)
Comment (isaacbrodsky) [reply src/h3lib/lib/algos.c:338]: Yes, that is my expectation

**Pull request merged**